gtsam/discrete/AlgebraicDecisionTree.h

```diff
@@ -138,6 +138,14 @@ namespace gtsam {
       return this->combine(labelC, &Ring::add);
     }
 
+    /// Equality method customized to node type `double`.
+    bool equals(const AlgebraicDecisionTree& other, double tol = 1e-9) const {
+      // lambda for comparison of two doubles upto some tolerance.
+      auto comparator = [](double a, double b, double tol) {
```

Comment:
Sorry, but this still does not make sense to me. I think the root issue is that in `DecisionTree.*` we have both tol and compare functions as arguments. I think it should be *only* the compare function, which should *not* have a tol argument. Instead, we should have here:
```
      auto compare = [tol](double a, double b) { return std::abs(a - b) < tol; };
      return this->root_->equals(*other.root_, compare);
```

Also, comparator -> compare. Former is weird.

Reply:
No worries about the iteration. We want this to be kickass! :slightly_smiling_face: 

Also that snippet helps, thank you!

Reply:
Also does this mean I should remove `tol` from the `equals` method and update the traits declarations accordingly?

Reply:
We removed Testable from DecisionTree, so not needed there.
For ADT, keep tol and traits if they exist.

Reply:
Ha guess not :smile: